plugins/chat.py

```diff
@@ -103,17 +106,26 @@ async def invite_link(message: Message):
     try:
         chat = await userge.get_chat(chat_id)
         chat_name = chat.title
-        if message.chat.type in ['group', 'supergroup', 'private']:
+        if chat.type in ['group', 'supergroup', 'private']:
             link = await userge.export_chat_invite_link(chat_id)
             await message.edit(
                 "**Invite link Genrated Successfully for\n"
                 f"{chat_name}**\n[Click here to join]({link})",
                 disable_web_page_preview=True)
         else:
-            await message.edit("```Rquirements not met...```")
+            await message.edit("```Requirements not met...```")
+    except ChatAdminRequired:
+        if chat.username:
+            await CHANNEL.log(f"You is not admin in @{chat.username}, can't Generate invite link...")
+            await message.edit(f"```You is not admin in @{chat.username}, can't Generate invite link (-_-)```")
+        else:
+            await CHANNEL.log(f"You is not admin in {chat.title}, can't Generate invite link...")
+            await message.edit(f"```You is not admin in {chat.title}, Can't Generate invite link (-.-)```")
+            return
```

Comment:
why doing same ?

```suggestion
        if chat.username:
            await message.edit(f"```You is not admin in @{chat.username}, can't Generate invite link (-_-)```", log=__name__)
        else:
            await message.edit(f"```You is not admin in {chat.title}, Can't Generate invite link (-.-)```", log=__name__)
            return
```

Reply:
Most probably to check if group is public and self is not admin so instead of taking out invite link provide username of public group. Else if group is private pass an error message 🤔🤔

Reply:
@Phyco-Ninja  yes
i just mentioned that duplicate thing